lib/editor/actions/map/stopStrategies.js

```diff
@@ -553,3 +562,276 @@ export function removeStopFromPattern (pattern: Pattern, stop: GtfsStop, index:
     dispatch(saveActiveGtfsEntity('trippattern'))
   }
 }
+
+/**
+ * Based on the provided starting point for deletion and the number of segments until we hit another stop,
+ * this function calculates the number of control points and adds the control points and segments to the array
+ * for later deletion.
+ */
+const updateControlPointsAndSegments = (deletedControlPoints: Array<number>, deletedSegments: Array<number>, deletionStartIndex: number, numberSegments: number): Array<Array<number>> => {
+  // Each control point creates 2 segments, so you have one less control point than segments.
+  const numberControlPoints = numberSegments - 1
+  // Add control points and segments to the list to delete and return
+  return [
+    addRangeToArray(deletedControlPoints, deletionStartIndex + 1, numberControlPoints), // Add one for immediate next control point.
+    addRangeToArray(deletedSegments, deletionStartIndex, numberSegments)
+  ]
+}
+
+/**
+ * Generates a new segment for the fromPoint and toPoint and inserts it into the new patternSegments array.
+ */
+const generateSegmentAndInsert = async (segments: Array<Coordinates>, fromPoint: ?ControlPoint, toPoint: ?ControlPoint, insertionPoint: number) => {
+  if (!fromPoint || !toPoint) return
+  const points = [fromPoint.point.geometry.coordinates, toPoint.point.geometry.coordinates]
+  const segment = await getSegment(points, true)
+  if (segment && segment.coordinates) segments.splice(insertionPoint, 0, segment.coordinates)
+}
+
+/**
+ * Updates the shape_dist_traveled values for each control point after a change in the pattern has occurred. These values must also be updated in the
+ * patternStops which will be used to create the shapeDistTraveled values in stop_times.txt
+ */
+const updateShapeDistTraveled = (controlPoints: Array<ControlPoint | null>, segments: Array<Coordinate>, patternStops: Array<PatternStop>): void => {
```

Comment:
I thought this PR was intended to address #941, but I am still able to reproduce that issue in this branch after following the steps described.

Reply:
This PR is designed to address an issue separate from #941 that has not been ticketed yet. It is a bug that occurs when you rearrange a pattern by dragging the PatternStopCards into a different order. Previously, this action would produce errors with the route shapes.